### PR TITLE
i#3170 accept custom [u]int64 definitions: fix undef ordering

### LINKED ISSUE #3198: opcode_mix_t::process_memref return statement(s) should be bools
In one place, we return a string.

### PR DESCRIPTION
**Pull request by mtrofin**
The DR_DO_NOT_DEFINE_xyz defines should be undefined before we end up inadvertently defining DR_DEFINE_FOR_xyz, otherwise DR_DO_NOT_DEFINE never take effect.

Issue: #3170

### CODE REVIEW

**Comment by derekbruening** [comment]
Description is incorrect.

**Comment by mtrofin** [comment]
Corrected description

File: core/lib/globals_shared.h [hunk 1]
```diff
@@ -211,6 +211,14 @@ typedef byte *app_pc;
```
Comment (derekbruening) [inline core/lib/globals_shared.h]: > Issue #3170
nit: missing :

**Comment by derekbruening** [comment]
> Is there a way to add a build test?

`git grep build-and-test`, but this should be much easier: just have a client target whose build flags set one of these defines.

File: suite/tests/CMakeLists.txt [hunk 2]
```diff
@@ -1706,6 +1706,7 @@ if (CLIENT_INTERFACE)
   include_directories("${PROJECT_BINARY_DIR}/suite/tests/annotations")
 
   tobuild_ci(client.call-retarget client-interface/call-retarget.c "" "" "")
+  tobuild_ci(int64_overrides client-interface/int64_overrides.c "" "" "")
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: Please match all of the peer tests: they have a `client.` prefix, so `client.int64-overrides`.

File: suite/tests/client-interface/int64_overrides.template [hunk 3]
```diff
@@ -0,0 +1,2 @@
+1
+Hello, world!
```
Comment (derekbruening) [inline suite/tests/client-interface/int64_overrides.template]: This should just a `.expect` file, right?  https://github.com/DynamoRIO/dynamorio/wiki/Test-Suite#test-output

File: suite/tests/client-interface/int64_overrides.dll.c [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+typedef char uint64;
```
Comment (derekbruening) [inline suite/tests/client-interface/int64_overrides.dll.c]: Please include a comment explaining what this is doing

File: suite/tests/CMakeLists.txt (hunk 2 above)
Comment (mtrofin) [reply suite/tests/CMakeLists.txt]: done

File: suite/tests/client-interface/int64_overrides.template (hunk 3 above)
Comment (mtrofin) [reply suite/tests/client-interface/int64_overrides.template]: Done

**Pull request merged**